pkg/section/local_module.go

```diff
@@ -33,27 +37,125 @@ func (m *LocalModule) Type() string {
 	return LocalModuleType
 }
 
-// Configure configures the module section by finding the module
-// for the current path
 func (m *LocalModule) Configure(path string) error {
 	if path != "" {
-		m.Path = path
-	} else {
-		path, err := findLocalModule()
+		m.Paths = []string{path}
+		return nil
+	}
+
+	modPaths, err := m.findLocalModules()
+	if err != nil {
+		return fmt.Errorf("unable to find local modules: %v", err)
+	}
+
+	if len(modPaths) == 0 {
+		return errors.New("could not find module path for `localModule` configuration")
+	}
+
+	m.Paths = modPaths
+	return nil
+}
+
+func (m *LocalModule) findLocalModules() ([]string, error) {
+	{ // try with one root go.mod
```

Comment:
Please do not use code blocks, a comment is enough.

Could change to search `go.work` first:
```
project/
├── go.work
├── myapp/
│   ├── go.mod         
│   └── main.go      
└── mylib/             
    ├── go.mod       
    └── lib.go
```